Error while using local C++ pointer in parallel loop

Error found in a machine with the following specifications

> Linux 64 bits (CentOS 6.9)
> Cython:  0.25.2 
> gcc :    6.4.0
> g++ :    6.4.0 (but not used a priori)
> GLIBC :  2.12 

Found the following error.

```
tofu/geom/_vignetting_tools.cpp:3034:55: erreur : invalid conversion from «int» to «double*» [-fpermissive]
                       double * __pyx_parallel_temp0 = 1;
                                                       ^
tofu/geom/_vignetting_tools.cpp:3036:52: erreur : invalid conversion from «int» to « int*» [-fpermissive]
                       int * __pyx_parallel_temp2 = 1;
                                                    ^
error: command 'gcc' failed with exit status 1
 
(base) [ DV226270 sirrah ~/ToFu_All/tofu_git/tofu (Issue61_Imas2tofuMoreRobust)] python setup.py build_ext --inplace

Version for setup.py :  1.4.0-338-g841d1f8

Compiling tofu/geom/_vignetting_tools.pyx because it changed.
[1/1] Cythonizing tofu/geom/_vignetting_tools.pyx

Error compiling Cython file:
------------------------------------------------------------
...
    with nogil, parallel(num_threads=num_threads):
        diff = NULL
        lref = NULL
        for ivign in prange(nvign):
            nvert = lnvert[ivign]
            diff = <double*>malloc(3*nvert*sizeof(double))
                ^
```

Compilation line (produced by setup.py):

`gcc -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/Applications/Anaconda/python36/lib/python3.6/site-packages/numpy/core/include -I/Applications/Anaconda/python36/include/python3.6m -c tofu/geom/_vignetting_tools.cpp -o build/temp.linux-x86_64-3.6/tofu/geom/_vignetting_tools.o -O0 -Wall -fopenmp`

After merging from 122_Pointer_Cpp to Issue61_Imas2TofuMoreRobust, on the IRFM servers I still get the following error at compilation:

`[ DV226270 spica ~/ToFu_All/tofu_git/tofu (Issue61_Imas2tofuMoreRobust)] python setup.py build_ext --inplace

Version for setup.py :  1.4.0-385-g4d0621f

your setuptools is too old (<12)
setuptools_scm functionality is degraded
/Applications/Anaconda/python36/lib/python3.6/site-packages/setuptools/dist.py:481: UserWarning: The version specified ('1.4.0-385-g4d0621f') is an invalid version, this may not work as expected with newer versions of setuptools, pip, and PyPI. Please see PEP 440 for more details.
  "details." % self.metadata.version
running build_ext
building 'tofu.geom._vignetting_tools' extension
gcc -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/Applications/Anaconda/python36/lib/python3.6/site-packages/numpy/core/include -I/Applications/Anaconda/python36/include/python3.6m -c tofu/geom/_vignetting_tools.cpp -o build/temp.linux-x86_64-3.6/tofu/geom/_vignetting_tools.o -O0 -Wall -fopenmp
cc1plus: warning: command line option \u2018-Wstrict-prototypes\u2019 is valid for C/ObjC but not for C++
tofu/geom/_vignetting_tools.cpp: In function \u2018int __pyx_f_4tofu_4geom_17_vignetting_tools_triangulate_polys(double**, long int*, int, long int**, int)\u2019:
tofu/geom/_vignetting_tools.cpp:3053:55: error: invalid conversion from \u2018int\u2019 to \u2018double*\u2019 [-fpermissive]
                       double * __pyx_parallel_temp0 = 1;
                                                       ^
tofu/geom/_vignetting_tools.cpp:3055:52: error: invalid conversion from \u2018int\u2019 to \u2018int*\u2019 [-fpermissive]
                       int * __pyx_parallel_temp2 = 1;
                                                    ^
error: command 'gcc' failed with exit status 1
`

--- a/tofu/geom/_vignetting_tools.pyx
+++ b/tofu/geom/_vignetting_tools.pyx
@@ -256,26 +256,25 @@ cdef inline int triangulate_polys(double** vignett_poly,
     cdef bint* lref
     # ...
     # -- Defining parallel part ------------------------------------------------
-    # with nogil, parallel(num_threads=num_threads):
-        # diff = NULL
-        # lref = NULL
-        # for ivign in prange(nvign):
-    for ivign in range(nvign):
-        nvert = lnvert[ivign]
-        diff = <double*>malloc(3*nvert*sizeof(double))
-        lref = <bint*>malloc(nvert*sizeof(bint))
-        ltri[ivign] = <long*>malloc((nvert-2)*3*sizeof(long))
-        if not diff or not lref or not ltri[ivign]:
-            with gil:
-                raise MemoryError()
-        try:
-            compute_diff3d(vignett_poly[ivign], nvert, &diff[0])
-            are_points_reflex(nvert, diff, &lref[0])
-            earclipping_poly(vignett_poly[ivign], &ltri[ivign][0],
-                             &diff[0], &lref[0], nvert)
-        finally:
-            free(diff)
-            free(lref)
+    with nogil, parallel(num_threads=num_threads):
+        diff = NULL
+        lref = NULL
+        for ivign in prange(nvign):
+            nvert = lnvert[ivign]
+            diff = <double*>malloc(3*nvert*sizeof(double))
+            lref = <bint*>malloc(nvert*sizeof(bint))
+            ltri[ivign] = <long*>malloc((nvert-2)*3*sizeof(long))
+            if not diff or not lref or not ltri[ivign]:
+                with gil:
+                    raise MemoryError()
+            try:
+                compute_diff3d(vignett_poly[ivign], nvert, &diff[0])
+                are_points_reflex(nvert, diff, &lref[0])
+                earclipping_poly(vignett_poly[ivign], &ltri[ivign][0],
+                                 &diff[0], &lref[0], nvert)
+            finally:
+                free(diff)
+                free(lref)
 
     return 0
 

--- a/tofu/geom/_vignetting_tools.pyx
+++ b/tofu/geom/_vignetting_tools.pyx
@@ -252,13 +252,11 @@ cdef inline int triangulate_polys(double** vignett_poly,
     """
     cdef int ivign, ii
     cdef int nvert
-    cdef double* diff
-    cdef bint* lref
+    cdef double* diff = NULL
+    cdef bint* lref = NULL
     # ...
     # -- Defining parallel part ------------------------------------------------
     with nogil, parallel(num_threads=num_threads):
-        diff = NULL
-        lref = NULL
         for ivign in prange(nvign):
             nvert = lnvert[ivign]
             diff = <double*>malloc(3*nvert*sizeof(double))

